internal/promotion/registry.go

```diff
@@ -24,3 +26,7 @@ var stepRunnerReg = stepRunnerRegistry{}
 func RegisterStepRunner(runner promotion.StepRunner) {
 	stepRunnerReg.register(runner)
 }
+
+func GetStepRunner(step *Step) promotion.StepRunner {
```

Comment:
Might be better to simply do:

```suggestion
func GetStepRunner(name string) promotion.StepRunner {
```